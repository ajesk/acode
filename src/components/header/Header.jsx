import React from 'react';
import './Header.scss';

const Header = ({ setPage, pages }) => {
  return (
    <div className="header">
      <button className="header-name" onClick={() => setPage(0)}>Aaron Jeskie</button>
      <div className="vr" />
      {
        pages.filter(page => page.name !== '')
          .map((page, i) =>
            <button key={i} className="page-tab" onClick={() => setPage(page.id)}>
              {page.name}
            </button>)
      }
    </div>
  );
}

export default Header;
