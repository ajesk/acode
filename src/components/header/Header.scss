$background-color: #343a40;
$text-color: white;

.header {
  display: flex;
  align-content: center;

  flex: 0 1;

  background-color: $background-color;
  color: $text-color;
}

button:focus {
  color: invert($background-color);
  background-color: invert($text-color);
}

.header-name {  
  display: flex;
  justify-content: center;
  align-items: center;
  
  font-weight: bold;
  font-size: larger;
  color: inherit;
  border: 0;
  background-color: inherit;
  margin: 0.5em;
}

.vr {
  border-left: 1px solid;
}

.page-tab {
  flex: 0 1 auto;

  border: 0;
  border-right: 1px solid;
  padding: 0.5em;
  margin: 0.25em 0;

  font-size: large;
  font-weight: bold;
  color: $text-color;
  background-color: $background-color;
}