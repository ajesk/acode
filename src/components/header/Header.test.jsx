import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';

describe('header', () => {
  const props = {
    pages: [
      {name: "adaf", id: 1},
      {name: "xyz", id: 2},
      {name: "mno", id: 3},
      {name: "page", id: 4}
    ],
    setPage: () => ''
  };
  test('snapshot', () => {
    expect(render(<Header {...props} />)).toMatchSnapshot();
  });
});
