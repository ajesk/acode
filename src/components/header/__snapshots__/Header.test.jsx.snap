// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <button
          class="header-name"
        >
          Aaron Jeskie
        </button>
        <div
          class="vr"
        />
        <button
          class="page-tab"
        >
          adaf
        </button>
        <button
          class="page-tab"
        >
          xyz
        </button>
        <button
          class="page-tab"
        >
          mno
        </button>
        <button
          class="page-tab"
        >
          page
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <button
        class="header-name"
      >
        Aaron Jeskie
      </button>
      <div
        class="vr"
      />
      <button
        class="page-tab"
      >
        adaf
      </button>
      <button
        class="page-tab"
      >
        xyz
      </button>
      <button
        class="page-tab"
      >
        mno
      </button>
      <button
        class="page-tab"
      >
        page
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
